src/ethereum/london/spec.py

```diff
@@ -0,0 +1,1136 @@
+        chain.blocks = chain.blocks[-255:]
+
+
+def validate_header(header: Header, parent_header: Header) -> None:
+    """
+    Verifies a block header.
+
+    In order to consider a block's header valid, the logic for the
+    quantities in the header should match the logic for the block itself.
+    For example the header timestamp should be greater than the block's parent
+    timestamp because the block was created *after* the parent block.
+    Additionally, the block's number should be directly folowing the parent
+    block's number since it is the next block in the sequence.
+
+    Parameters
+    ----------
+    header :
+        Header to check for correctness.
+    parent_header :
+        Parent Header of the header to check for correctness
+    """
+    if header.number == MAINNET_FORK_BLOCK:
+        parent_gas_limit = parent_header.gas_limit * ELASTICITY_MULTIPLIER
+        parent_gas_target = parent_header.gas_limit
+    else:
+        parent_gas_limit = parent_header.gas_limit
+        parent_gas_target = parent_header.gas_limit // ELASTICITY_MULTIPLIER
+
+    parent_base_fee_per_gas = parent_header.base_fee_per_gas
+    parent_gas_used = parent_header.gas_used
+
+    ensure(header.gas_used <= header.gas_limit, InvalidBlock)
+    ensure(header.gas_limit >= 5000, InvalidBlock)
+    ensure(
+        header.gas_limit < parent_gas_limit + parent_gas_limit // 1024,
+        InvalidBlock,
+    )
+    ensure(
+        header.gas_limit > parent_gas_limit - parent_gas_limit // 1024,
+        InvalidBlock,
+    )
+
+    # check if the base fee is correct
+    if header.number == MAINNET_FORK_BLOCK:
+        expected_base_fee_per_gas = INITIAL_BASE_FEE
+    elif parent_gas_used == parent_gas_target:
+        expected_base_fee_per_gas = parent_base_fee_per_gas
+    elif parent_gas_used > parent_gas_target:
+        gas_used_delta = parent_gas_used - parent_gas_target
+        base_fee_per_gas_delta = max(
+            parent_base_fee_per_gas
+            * gas_used_delta
+            // parent_gas_target
+            // BASE_FEE_MAX_CHANGE_DENOMINATOR,
+            1,
+        )
+        expected_base_fee_per_gas = (
+            parent_base_fee_per_gas + base_fee_per_gas_delta
+        )
+    else:
+        gas_used_delta = parent_gas_target - parent_gas_used
+        base_fee_per_gas_delta = (
+            parent_base_fee_per_gas
+            * gas_used_delta
+            // parent_gas_target
+            // BASE_FEE_MAX_CHANGE_DENOMINATOR
+        )
+        expected_base_fee_per_gas = (
+            parent_base_fee_per_gas - base_fee_per_gas_delta
+        )
+
+    ensure(expected_base_fee_per_gas == header.base_fee_per_gas, InvalidBlock)
+
+    parent_has_ommers = parent_header.ommers_hash != EMPTY_OMMER_HASH
+    ensure(header.timestamp > parent_header.timestamp, InvalidBlock)
+    ensure(header.number == parent_header.number + 1, InvalidBlock)
+    ensure(
+        check_gas_limit(header.gas_limit, parent_header.gas_limit),
```

Comment:
In the transition block you must compare `header.gas_limit` to `2 * parent_header.gas_limit`

Reply:
As it turns out, this was already being checked before this line. I have removed the redundancy and moved this check up to where the EIP specifies it.